src/core/graph.js

```diff
@@ -535,12 +703,29 @@ export class Graph {
     return result;
   }
 
+  /**
+   * Produce a copy of a Graph.
+   *
+   * The copy is equal to the original, but distinct, so that they may be
+   * modified independently.
+   */
   copy(): Graph {
     const result = Graph.merge([this]);
     this._maybeCheckInvariants();
     return result;
   }
 
+  /**
+   * Serialize a Graph into JSON.
```

Comment:
```
   * Serialize a Graph into JSON.
```

Ambiguous: “JSON” could well be interpreted to mean a string rather than
what you mean, which is a “plain old JavaScript object” or “JSON
object”.

```
   *
   * We store the nodes in sorted order. This allows us to serialize the edges
   * in an indexed representation, where each edge has an address and then
   * represents its src and dst by sorted node indices.
   *
   * This greatly reduces the size of the representation, and means that this
   * method runs in O(n log n + e) where n is the number of nodes and e is the
   * number of edges.
```

These are implementation details and should _not_ be documented. If we
want an API that exposes the performance benefits of this representation
to other clients, then that is a totally separate matter.

Suggest replacing with:

> Runs in time `O(n log n + e)`, where `n` is the number of nodes and
> `e` is the number of edges.